File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        int sum = 0;
        for(int num : nums) sum += num;
        int len = -1, currSum = 0;
        for(int l=0, r=0; r<nums.size(); r++){
            currSum += nums[r];
            while(l <= r && currSum>sum-x) currSum -= nums[l++];
            if(currSum == sum-x) len = max(len, r-l+1);
        }
        return len==-1 ? -1 : nums.size()-len;
    }
};

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        totalSum = sum(nums)
        l, r, currSum, length = 0, 0, 0, -1
        while r < len(nums):
            currSum += nums[r]
            while l<=r and currSum>totalSum-x:
                currSum -= nums[l]
                l += 1
            if currSum == totalSum-x: length = max(length, r-l+1)
            r += 1
        return -1 if length==-1 else len(nums)-length